pytket/binders/passes.cpp

```diff
@@ -145,6 +145,31 @@ const PassPtr &DecomposeClassicalExp() {
   return pp;
 }
 
+std::optional<OpTypeSet> get_gate_set(const BasePass &base_pass) {
+  OpTypeSet allowed_ops;
+  for (const std::pair<const std::type_index, std::shared_ptr<tket::Predicate>>
+           &p : base_pass.get_conditions().first) {
+    if (p.second->to_string().substr(0, 17) == "GateSetPredicate:") {
```

Comment:
I don't think this check is necessary. We can just use `dynamic_cast` and ignore it if it returns `nullptr`.